core/src/main/java/hudson/util/io/Archiver.java

```diff
@@ -36,11 +38,16 @@
  */
 public abstract class Archiver extends FileVisitor implements Closeable {
     protected int entriesWritten =0;
+    protected List<String> files = new ArrayList<>();
 
     /**
      * Number of files/directories archived.
      */
     public int countEntries() {
         return entriesWritten;
     }
+
+    public List<String> files(){
```

Comment:
not needed.  should be the job of any `Consumer` to actually record this if it is wanted.